### PR TITLE
SDCICD-56: Migrate to go modules

### PR DESCRIPTION
**Pull request by jeefy**
This is the initial work needed to migrate our dependencies to go modules. I've tested this locally with success 🎉 but want more eyes 👀on this before we consider migrating.

/hold
/assign @meowfaceman

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,37 @@
+module github.com/openshift/osde2e
+
+go 1.12
```
Comment (meowfaceman) [inline go.mod]: I'm guessing this would need to change if we moved to the most recent go version.

**Comment by jeremyeder** [comment]
Just a sec - the base image here is a debian image from dockerhub?  We should be using an RH base image.  For example registry.svc.ci.openshift.org/ocp/builder:golang-1.12 ?

**Comment by jeefy** [comment]
/hold cancel

ready for lgtm

**Comment by jeefy** [comment]
Oooh one sec, just saw that comment @jeremyeder

**Comment by jeefy** [comment]
@jeremyeder Thoughts on using the alpine tag? We're building then copying artifacts into `distroless` anyhow.

**Comment by jeremyeder** [comment]
Use RH container images please. (openshift policy)

**Comment by jeefy** [comment]
@jeremyeder @meowfaceman PTAL

File: Dockerfile [hunk 2]
```diff
@@ -1,24 +1,24 @@
-FROM golang:1.11.9
+FROM registry.svc.ci.openshift.org/openshift/release:golang-1.13
+
 ENV PKG=/go/src/github.com/openshift/osde2e/
+ENV GOMODULE111=on
```
Comment (meowfaceman) [inline Dockerfile]: We don't need this now, right?

Comment (jeefy) [reply Dockerfile]: You saw nothing

image

**Comment by jeefy** [comment]
Get credentials to the internal registry and `make build-image` :)

**Pull request merged**